lib/acme/resources/challenges/dvsni01.rb

```diff
@@ -0,0 +1,12 @@
+class Acme::Resources::Challenges::DVSNI01 < Acme::Resources::Challenges::Base
+
+  def sni_name
+    hd = crypto.digest.hexdigest(authorization_key)
+    "#{hd[0..31]}.#{hd[32..64]}.acme.invalid"
+  end
+
+  def request_verification
+    response = client.connection.post(@uri, { resource: 'challenge', type: 'tls-dvsni-01', keyAuthorization: authorization_key })
```

Comment:
It's `tls-sni-01`, not `tls-dvsni-01`. Honestly I think you need a test environment :/